src/components/useExchangeCode.js

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react';
+import { useQuery } from 'react-query';
+import { useIntl } from 'react-intl';
+
+import useOkapiKy from '../useOkapiKy';
+import {
+  getLoginTenant,
+} from '../loginServices';
+import { useStripes } from '../StripesContext';
+
+const useExchangeCode = () => {
+  const stripes = useStripes();
+  const ky = useOkapiKy();
+  const intl = useIntl();
+
+  const [error, setError] = useState();
+
+  const urlParams = new URLSearchParams(window.location.search);
+  const code = urlParams.get('code');
+  const { tenant, clientId } = getLoginTenant(stripes.okapi, stripes.config);
+
+  const { isFetching, data } = useQuery(
+    ['@folio/stripes-core', 'authn/token'],
```

Comment:
I would probably put a code in query key as well, we have cache config so there is small chance to get smth unexpected for 2 different tokens (IMO it's edge-edge case, but still)